File: app/src/main/java/net/squanchy/schedule/ScheduleModule.java
package net.squanchy.schedule;

import net.squanchy.service.firebase.FirebaseAuthService;
import net.squanchy.service.firebase.FirebaseDbService;
import net.squanchy.service.repository.EventRepository;

import dagger.Module;
import dagger.Provides;

@Module
class ScheduleModule {

    @Provides
    ScheduleService scheduleService(
            FirebaseDbService dbService,
            FirebaseAuthService authService,
            EventRepository eventRepository
    ) {
        return new ScheduleService(dbService, authService, eventRepository);
    }
}

File: app/src/main/java/net/squanchy/schedule/ScheduleModule.kt
package net.squanchy.schedule

import dagger.Module
import dagger.Provides
import net.squanchy.service.firebase.FirestoreDbService
import net.squanchy.service.repository.AuthService
import net.squanchy.support.checksum.ChecksumModule
import net.squanchy.support.checksum.Checksum

@Module(includes = [ChecksumModule::class])
class ScheduleModule {

    @Provides
    internal fun scheduleService(
        authService: AuthService,
        dbService: FirestoreDbService,
        tracksFilter: TracksFilter,
        checksum: Checksum
    ): ScheduleService = FirestoreScheduleService(authService, dbService, tracksFilter, checksum)
}
